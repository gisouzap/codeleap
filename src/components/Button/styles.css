.button {
  width: 112px;
  height: 32px;
  border-radius: 8px;
  border: none;
  cursor: pointer;
  transition: 150ms ease-in;
  font-weight: 700;
}

.button:hover {
  opacity: 0.8;
}

.button:active {
  opacity: 1;
}

.primary {
  font-size: 16px;
  line-height: 19px;
  color: #fff;
  text-transform: uppercase;
  background: var(--color-primary);
}

.button:disabled {
  background-color: var(--color-grey-100);
  cursor: not-allowed;
}

.button:disabled:hover {
  opacity: 1;
}

.danger {
  background: var(--color-danger);
  color: var(--color-white);
}

.light {
  border: 1px solid var(--color-grey-400);
  background-color: var(--color-white);
}

.success {
  background-color: var(--color-success);
  color: var(--color-white);
}
