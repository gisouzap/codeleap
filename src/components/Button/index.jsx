import PropTypes from 'prop-types';
import './styles.css';

function Button({ label, variant, disabled, ...props }) {
  return (
    <button
      type="button"
      disabled={disabled}
      {...props}
      className={`button ${variant}`}
    >
      {label}
    </button>
  );
}

Button.propTypes = {
  label: PropTypes.string.isRequired,
  disabled: PropTypes.bool,
  variant: PropTypes.string.isRequired,
  children: PropTypes.node,
};

export default Button;
