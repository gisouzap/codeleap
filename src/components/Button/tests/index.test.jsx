import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import Button from '../index';

describe('Button', () => {
  it('should render button with correct label', () => {
    const { getByRole } = render(<Button label="Confirm" variant="primary" />);

    const button = getByRole('button');

    expect(button).toBeInTheDocument();
  });

  it('should call handleClick when the button is clicked', () => {
    const handleClick = jest.fn();

    const { getByRole } = render(
      <Button label="Confirm" onClick={handleClick} variant="primary" />
    );

    const button = getByRole('button');
    fireEvent.click(button);
    expect(handleClick).toHaveBeenCalledTimes(1);
  });

  it('should render and match the snapshot', () => {
    const { container } = render(<Button label="Confirm" variant="primary" />);

    expect(container).toMatchSnapshot();
  });
});
