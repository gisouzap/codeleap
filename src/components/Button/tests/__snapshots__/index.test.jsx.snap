// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render and match the snapshot 1`] = `
<div>
  <button
    class="button primary"
    type="button"
  >
    Confirm
  </button>
</div>
`;
